### PR TITLE
Generate Variations

### PR DESCRIPTION
**Pull request by fe-lix-**
Extension to add Variation Generation to Universal Editor

## Description

Extension to add Variation Generation to Universal Editor

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: universal-editor-generation-variation/app.config.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+extensions:
+  aem/universal-editor/1:
```
Comment (slitviachenko) [inline universal-editor-generation-variation/app.config.yaml:2]: @fe-lix- , It looks like the `aem/universal-editor/1` extension point was renamed to `universal-editor/ui/1`, see [this slack thread](https://cq-dev.slack.com/archives/C03RRV5M9B9/p1695059575115709) for more details.

File: universal-editor-generation-variation/src/aem-universal-editor-1/web-src/src/components/ExtensionRegistration.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        headerMenu: {
+          getButtons() {
+            return [
+              // @todo YOUR HEADER BUTTONS DECLARATION SHOULD BE HERE
+              {
+                id: 'generate-variations',
+                label: 'Generate Variations',
+                icon: 'OpenIn',
+                onClick() {
+                  window.open(generationVariationURL, '_blank')
```
Comment (slitviachenko) [inline universal-editor-generation-variation/src/aem-universal-editor-1/web-src/src/components/ExtensionRegistration.js:25]: @fe-lix- , Is it enough to redirect a customer to this URL? It looks like some additional info should be provided via query parameters, like `ref`, `repo`, `owner`. Not sure if Universal Editor provides it in the shared context to UI extensions.

**Review by slitviachenko — CHANGES_REQUESTED** [review]
@fe-lix- , left some comments

File: universal-editor-generation-variation/src/aem-universal-editor-1/web-src/src/components/ExtensionRegistration.js (hunk 2 above)
Comment (slitviachenko) [reply universal-editor-generation-variation/src/aem-universal-editor-1/web-src/src/components/ExtensionRegistration.js:25]: For me this URL doesn't work, in console output I see
```
Uncaught Error: It seems we're missing the ref, repo or owner search parameter in your application.
    at getWebsiteUrl ...
```

Comment (fe-lix-) [reply universal-editor-generation-variation/src/aem-universal-editor-1/web-src/src/components/ExtensionRegistration.js:25]: Currently the button does not work, Experience Success team they would be able to be "context free" for Summit.

File: universal-editor-generation-variation/app.config.yaml (hunk 1 above)
Comment (fe-lix-) [reply universal-editor-generation-variation/app.config.yaml:2]: Thanks!

File: universal-editor-generation-variation/src/universal-editor-ui-1/web-src/src/components/ExtensionRegistration.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        headerMenu: {
+          getButtons() {
+            return [
+              // @todo YOUR HEADER BUTTONS DECLARATION SHOULD BE HERE
+              {
+                id: 'generate-variations',
+                label: 'Generate Variations',
+                icon: 'OpenIn',
+                onClick() {
+                  window.open(generationVariationURL, '_blank')
```
Comment (fe-lix-) [inline universal-editor-generation-variation/src/universal-editor-ui-1/web-src/src/components/ExtensionRegistration.js:25]: @naydav Will the target `_blank` be an issue in Universal Editor? (cc @slitviachenko )

Comment (slitviachenko) [reply universal-editor-generation-variation/src/universal-editor-ui-1/web-src/src/components/ExtensionRegistration.js]: On the latest UE main branch using Chrome I got the following error
Screenshot 2024-02-25 at 9 19 20 PM

Comment (slitviachenko) [reply universal-editor-generation-variation/src/universal-editor-ui-1/web-src/src/components/ExtensionRegistration.js]: @fe-lix- , I used a local UE setup. Didn't Chrome block pop-ups from extensions? For me it did by default, I have an option to explicitly allow pop-ups, but that is another story. I'll check with stage, prod UE then.

Comment (fe-lix-) [reply universal-editor-generation-variation/src/universal-editor-ui-1/web-src/src/components/ExtensionRegistration.js]: I had a chat with the Editor XP team there is no pop-up protection in Universal Editor, it's a per browser configuration.

File: universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/index.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (lenaorobei) [inline universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/index.js:1]: This file can be as simple as

``` js
import "core-js/stable";

window.React = require("react");
import ReactDOM from "react-dom";

import App from "./components/App";

ReactDOM.render(<App />, document.getElementById("root"));
```

File: universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/exc-runtime.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/*
```
Comment (lenaorobei) [inline universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/exc-runtime.js:1]: This code is not being used

File: universal-editor-generate-variation/jest.setup.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/*
```
Comment (lenaorobei) [inline universal-editor-generate-variation/jest.setup.js:1]: Is this needed?

File: universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/components/ExtensionRegistration.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+function ExtensionRegistration() {
+  const init = async () => {
+    const guestConnection = await register({
+      id: extensionId,
+      methods: {
+        headerMenu: {
+          getButtons() {
+            return [
+              // @todo YOUR HEADER BUTTONS DECLARATION SHOULD BE HERE
```
Comment (lenaorobei) [inline universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/components/ExtensionRegistration.js:19]: Please remove `@todo`

File: universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/components/Spinner.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+/*
+ * <license header>
```
Comment (lenaorobei) [inline universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/components/Spinner.js:2]: Missing license

File: universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/index.css [hunk 9]
```diff
@@ -0,0 +1,9 @@
+/*
+* <license header>
```
Comment (lenaorobei) [inline universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/index.css:2]: Please fix header

File: universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/utils.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @param  {string} actionUrl
+ * @param {object} headers
+ * @param  {object} params
+ *
+ * @returns {Promise<string|object>} the response
+ *
+ */
+
+async function actionWebInvoke (actionUrl, headers = {}, params = {}, options = { method: 'POST' }) {
```
Comment (lenaorobei) [inline universal-editor-generate-variation/src/universal-editor-ui-1/web-src/src/utils.js:19]: We don't have actions in this example.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please see my comments.

**Comment by lenaorobei** [comment]
@fe-lix- I'm closing this PR due to inactivity. If you plan to continue working on it, feel free to reopen. Thanks!

**Pull request closed without merging**